sqlalchemy-stubs/engine/interfaces.pyi

```diff
@@ -1,214 +1,315 @@
     ) -> None: ...
     def do_execute_no_params(
         self,
-        cursor: Any,
+        cursor: _DBAPICursor,
         statement: Any,
         parameters: Any,
-        context: Optional[Any] = ...,
+        context: Optional[ExecutionContext] = ...,
+    ) -> None: ...
+    def is_disconnect(
+        self, e: Any, connection: Connection, cursor: _DBAPICursor
+    ) -> bool: ...
+    def connect(self, *cargs: Any, **cparams: Any) -> Any: ...
+    def on_connect_url(self, url: URL) -> Optional[_OnConnect]: ...
+    def on_connect(self) -> Optional[_OnConnect]: ...
+    def reset_isolation_level(self, dbapi_conn: _DBAPIConnection) -> None: ...
+    def set_isolation_level(
+        self, dbapi_conn: _DBAPIConnection, level: Any
     ) -> None: ...
-    def is_disconnect(self, e: Any, connection: Any, cursor: Any) -> None: ...
-    def connect(self, *cargs: Any, **cparams: Any) -> None: ...
-    def on_connect(self) -> None: ...
-    def reset_isolation_level(self, dbapi_conn: Any) -> None: ...
-    def set_isolation_level(self, dbapi_conn: Any, level: Any) -> None: ...
-    def get_isolation_level(self, dbapi_conn: Any) -> None: ...
-    def get_default_isolation_level(self, dbapi_conn: Any) -> None: ...
+    def get_isolation_level(self, dbapi_conn: _DBAPIConnection) -> Any: ...
+    def get_default_isolation_level(
+        self, dbapi_conn: _DBAPIConnection
+    ) -> Any: ...
     @classmethod
-    def get_dialect_cls(cls, url: Any): ...
+    def get_dialect_cls(cls, url: URL) -> Dialect: ...
     @classmethod
     def load_provisioning(cls) -> None: ...
     @classmethod
-    def engine_created(cls, engine: Any) -> None: ...
+    def engine_created(cls, engine: Engine) -> None: ...
 
 class CreateEnginePlugin:
-    url: Any = ...
-    def __init__(self, url: URL, kwargs: dict[str, Any]) -> None: ...
-    def update_url(self, url: Any) -> None: ...
+    url: URL = ...
+    def __init__(self, url: URL, kwargs: Dict[str, Any]) -> None: ...
+    def update_url(self, url: URL) -> URL: ...
     def handle_dialect_kwargs(
-        self, dialect_cls: Any, dialect_args: Any
+        self, dialect_cls: Type[Dialect], dialect_args: Dict[str, Any]
     ) -> None: ...
-    def handle_pool_kwargs(self, pool_cls: Any, pool_args: Any) -> None: ...
-    def engine_created(self, engine: Any) -> None: ...
+    def handle_pool_kwargs(
+        self, pool_cls: Any, pool_args: Dict[str, Any]
+    ) -> None: ...
+    def engine_created(self, engine: Engine) -> None: ...
 
 class ExecutionContext:
-    def create_cursor(self) -> None: ...
+    connection: Connection = ...
+    root_connection: Connection = ...
+    dialect: Dialect = ...
+    cursor: _DBAPICursor = ...
+    compiled: Optional[Compiled] = ...
+    statement: Optional[str] = ...
+    parameters: Any = ...
+    isinsert: bool = ...
+    isupdate: bool = ...
+    should_autocommit: bool = ...
+    prefetch_cols: bool = ...
+    postfetch_cols: bool = ...
```

Comment:
these seem list/sequences not bool https://github.com/sqlalchemy/sqlalchemy/blob/dd6f4543bc8ccbf07bfc5c8fb850be60ab420b57/lib/sqlalchemy/engine/interfaces.py#L1350-L1358